File: benchmarks/oopsla25/subjects/104.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f104(const char* s) {
    int i = strlen(s);
    while (i > 0) {
        assert(s[i - 1] == 'a');
        i--;
    }
    assert(strlen(s) == 2);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f104(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/104.py
#!/usr/bin/env python3

import sys

def f104(s: str):
  i = len(s)
  while i > 0:
    assert s[i-1] == "a"
    i -= 1
  assert len(s) == 2

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f104(my_string)
